### PR TITLE
Dark Mode Improvements

### LINKED ISSUE #5656: Unreadable color on dark mode
### Steps to reproduce
1. Change to dark theme
2. Go back to main screen

### Expected behaviour
- HeaderBar and some other components should display text with with or a light color on dark blue.

### Actual behaviour
- HeaderBar and some other components display text in black on dark blue making difficult to read.

### Environment data
Android version: 9

Device model: Redmi note 7

Stock or customized system: Stock

Nextcloud app version: 20200311 (Dev branch)

Nextcloud server version: 17

#### Web server error log
```
N/A
```

#### Nextcloud log (data/nextcloud.log)
```
N/A
```

### Screenshot
Screenshot_2020-03-13-13-22-38-157_com nextcloud android beta

### LINKED ISSUE #5657: Dark mode results in dark-on-dark elements only visible at high screen brightness
Submitting this as requested by the developer in response to Play review.

### Steps to reproduce
1. Set dark mode
2. Set brightness lower than half way
3. Look at the app

### Expected behaviour
- Setting dark mode should apply a dark theme, but any control elements (menus buttons, action buttons, button labels) should have lighter elements so the user can see where they are and what they do.

### Actual behaviour
- Menu controls and button labels are dark, so cannot be seen. Some not at all, others only when the screen is set to very high or maximum brightness which is not desirable in low light conditions. 

### Environment data
Android version: 8.1.0

Device model: Samsung S9

Stock or customized system: LineageOS 15.1

Nextcloud app version: 3.10.1

Nextcloud server version: 18.02 (affects previous versions too)

top right and top left menus invisible
1 - top right and top left menus invisible

keep and delete button text only visible at very high brightness
2 - keep and delete button text

dark icons on dark background only visible with very high brightness
3 - dark on dark only visible with high brightness

"cancel" action button text only visible with very high brightness.
4 - cancel text only visible with high brightness

### PR DESCRIPTION
**Pull request by dan0xii**
Removes conflicts between high contrast server themes and the dark mode.

When finished it will ~~close~~ #5657 and close #5656 and will possibly have some overlap with #5353.

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@dan0xii are you planning to add (much) more changes to the PR? Just asking since including such changes in 3.11.1 would be nice (already too late for 3.11.0).

Also looping in @tobiasKaminsky since this should make it into the first patch release in my opinion.

**Comment by dan0xii** [comment]
> @dan0xii are you planning to add (much) more changes to the PR? J

Yes, this only covers off the actionbar title colour--there's still a decent bit of work to try and make ThemeUtils more aware of conflicts between the dark theme and a server dark theme.

File: src/main/java/com/owncloud/android/utils/ThemeUtils.java [hunk 1]
```diff
@@ -143,6 +143,11 @@ public static int primaryColor(Account account, boolean replaceWhite, Context co
         }
     }
 
+    public static int getNeutralGrey(Context context) {
+        return darkTheme(context) ? context.getResources().getColor(R.color.fg_contrast) :
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/ThemeUtils.java]: if fg_contrast is also grey, please rename it to grey…

**Comment by tobiasKaminsky** [comment]
> (already too late for 3.11.0).

I fear that we need a RC3 because of conflict handling problems (I will create a PR later).
So if this is done by today, I can include it into RC3.

**Comment by dan0xii** [comment]
I've made a couple of quick changes because I didn't have a lot of time but made a bit of progress. (Hopefully I didn't break anything..)


Screenshot_1584393499

Screenshot_1584393513

Screenshot_1584394068

Screenshot_1584394077

**Comment by tobiasKaminsky** [comment]
I fixed upload a bit, but having white/black as primary color is a difficult case on light/dark mode.
So let us get this in as it is already an improvement :+1:

**Comment by dan0xii** [comment]
Rebased against master. Do you want me to stop making changes to this branch ?

**Comment by tobiasKaminsky** [comment]
> Rebased against master. Do you want me to stop making changes to this branch ?

How many changes / how long do you think you might need?
I want to create RC4 this afternoon, in 2-3h?

**Comment by dan0xii** [comment]
I'm done. I pushed my last change which I think closes #5656.

**Comment by tobiasKaminsky** [comment]
Then I'll review merge and backport it so that it is in RC4.
Thanks for your amazing work!

**Pull request merged**